crates/bevy_render/macros/src/render_ecs_resource.rs

```diff
@@ -0,0 +1,24 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, parse_quote, DeriveInput, Path};
+
+pub fn derive_extract_resource(input: TokenStream) -> TokenStream {
+    let mut ast = parse_macro_input!(input as DeriveInput);
+    let bevy_render_path: Path = crate::bevy_render_path();
+
+    ast.generics
+        .make_where_clause()
+        .predicates
+        .push(parse_quote! { Self: Clone });
+
+    let struct_name = &ast.ident;
+    let (impl_generics, type_generics, where_clause) = &ast.generics.split_for_impl();
+
+    TokenStream::from(quote! {
+        impl #impl_generics #bevy_render_path::render_ecs_resource::ExtractResource for #struct_name #type_generics #where_clause {
```

Comment:
One thing I didn't manage to figure out was how to make `#bevy_render_path` be either `crate` or `bevy_render` depending on whether it was being derived on a type within `bevy_render` or outside of it.

Reply:
It looks like you can add `extern crate self as bevy_render;` in `lib.rs` or so. I'm not sure if that's a good idea.

Reply:
@mockersf gave me this suggestion:
```rust
pub(crate) fn bevy_render_path() -> syn::Path {
    BevyManifest::default()
        .maybe_get_path("bevy_render")
        // NOTE: If the derivation is within bevy_render, then we need to return 'crate'
        .unwrap_or_else(|| BevyManifest::parse_str("crate"))
}
```